pkg/envinfo/envinfo.go

```diff
@@ -135,6 +138,11 @@ func newProxyEnvInfo() proxyEnvInfo {
 func (esi *EnvSpecificInfo) SetConfiguration(parameter string, value interface{}) (err error) {
 	if parameter, ok := asLocallySupportedParameter(parameter); ok {
 		switch parameter {
+		case "create":
```

Comment:
Why we use a single parameter for `Name`, `Type` and `Namespace`? 
Should these 3 be separate parameters?
Putting multiple settings into a signle parameter would cause error for some of the functions. e.g. `isSet()` returns `util.IsSet(esi.componentSettings, parameter)`.

Reply:
Single parameter can help assign all env fields related with `odo create` with one function call of `SetConfiguration` instead of multiple function calls of `SetConfiguration`, so using single parameter can:
- Save the number of function calls
- To make it clear that we get `Name`, `Type` and `Namespace` together from `odo create` and assign them together

Given the `SetConfiguration` is just for feeding info into `componentSettings` struct and write to local file system, so there shouldn't be any affect for single parameter.

Regarding the affect of `isSet()`, from the screenshot below we can see `isSet()` is only called by test cases, there is no code in product calls that function. (If that's the case I would suggest to put `isSet()` in `envinfo_test.go` instead of `envinfo.go`)
![Screen Shot 2020-03-11 at 2 40 20 PM](https://user-images.githubusercontent.com/15131537/76452928-474b6380-63a8-11ea-96ce-cc4bac436e55.png)

So `isSet()` is just a testing function to test if the `componentSettings` struct is set properly, using single parameter (set all fields together) shouldn't affect the setting of `componentSettings` struct. However, `util.IsSet(esi.componentSettings, parameter)` is the wrapper of [reflect package](https://golang.org/pkg/reflect/) which helps checking the field's kind, value, etc. The `util.IsSet(esi.componentSettings, parameter)` function itself is lack of checking `string` field, so I add the changes here https://github.com/openshift/odo/pull/2699/files#diff-975cb50bfa08efda085ba94fac1c40c9R81 and no errors will be caused.

Reply:
Understood. 
The `isSet()` function allows check of the parameter existence with env.yaml file, it is not a test function specific, even it's currently only being called in it's unit test.

Reply:
Make senses, we can keep it in `envinfo.go` for future usage.